### PR TITLE
Add Solution "Performance Analysis and Optimization"

### PR DESCRIPTION
**Pull request by fia0**

### CODE REVIEW

File: task3/checkpoint_fsync.c [hunk 1]
```diff
@@ -78,6 +82,13 @@ init_matrix (double** matrix)
 	}
 }
 
+inline long nano_diff(long l, long r) {
```
Comment (Cookie42) [inline task3/checkpoint_fsync.c]: This is only called at two points, the `l > r` will always be false in these cases. 
I would suggest to let this take the `timespec` structs and compute the counter increase here, will make it more readable.

Comment (fia0) [reply task3/checkpoint_fsync.c]: The `l > r` condition can be met for operations taking longer than one second. Which could happen in file synchronization, just a possibility I didn't want to forego.
You're right with the timespec, we can encapsulate some non-nice code there from our calculation function. Will move.

File: task3/checkpoint_fsync.c [hunk 2]
```diff
@@ -94,11 +94,20 @@ init_matrix (double** matrix)
...
+		foo = ((unsigned long long)(l.tv_sec - r.tv_sec)) * SEC_TO_NANO;
 	}
-	return r - l;
+	unsigned long long rest;
+	if (l.tv_nsec > r.tv_nsec) {
+		rest = l.tv_nsec - r.tv_nsec;
+	} else {
+		rest = r.tv_nsec - l.tv_nsec;
+	}
+	return foo + rest;
```
Comment (Cookie42) [inline task3/checkpoint_fsync.c]: But that is not how time works. Lets make an example, for simplicity assume `SEC_TO_NANO = 1000` : `l = {tv_sec = 0, tv_nsec = 200}` and `r = {tv_sec = 1, tv_nsec = 100}`. We should see that the diff are 900 nanos. 
This method currently makes `foo=1000` and then computes `rest=100`. So the overall time is 1100, which is just wrong.
It should always be `end timestamp - start timestamp`. The rest can and should be signed.

**Review by Cookie42 — CHANGES_REQUESTED** [review]

File: task3/checkpoint_fsync.c (hunk 2 above)
Comment (fia0) [reply task3/checkpoint_fsync.c]: True, yeah sorry was in a hurry there and totally did something else, you are absolutely right. Gonna correct this right away.

**Review by Cookie42 — APPROVED** [review]

**Comment by fia0** [comment]
Merging the current version there is not much time left!

**Comment by fia0** [comment]
Everything is done so far, just quckly wrote the remaining parts of the last task

**Pull request merged**